docs/about/contributing/modules.md

```diff
@@ -0,0 +1,449 @@
+  })
+}
+```
+
+**Error handling** in your shell scripts:
+
+```bash
+#!/bin/bash
+set -euo pipefail
+
+# Platform detection
+case "${ARCH}" in
+  "amd64") ARCH_SUFFIX="" ;;
+  "arm64") ARCH_SUFFIX="-arm64" ;;
+  *) echo "ERROR: Unsupported architecture: ${ARCH}"; exit 1 ;;
+esac
+
+# Tool installation with error checking
+if ! command -v tool &> /dev/null; then
+    echo "Installing tool..."
+    # Installation commands
+fi
+```
+
+### Integration Patterns
+
+**Conditional resources** using `start_count`:
+
+```tf
+resource "coder_app" "service" {
+  count    = data.coder_workspace.me.start_count
+  agent_id = var.agent_id
+  # Only created when workspace is running
+}
+```
+
+**Multi-step installation** with dependencies:
+
+```tf
+resource "coder_script" "install_deps" {
+  agent_id = var.agent_id
+  script   = "install_dependencies.sh"
+  order    = 1
+}
+
+resource "coder_script" "configure_service" {
+  agent_id   = var.agent_id
+  script     = "configure_service.sh"
+  depends_on = [coder_script.install_deps]
+  order      = 2
+}
+```
+
+**Health checks** for web services:
+
+```tf
+resource "coder_app" "service" {
+  agent_id = var.agent_id
+  url      = "http://localhost:${var.port}"
+  
+  healthcheck {
+    url       = "http://localhost:${var.port}/health"
+    interval  = 5
+    threshold = 6
+  }
+}
+```
+
+## Testing and Documentation
+
+### Module Testing
+
+While not mandatory, **tests are recommended** for all modules. A `main.test.ts` file validates that the module works correctly. Tests use **Bun** as the test runner and verify that:
+
+- **Custom configurations** work with different variable values
+- **Default configuration** creates expected resources
+- **Input validation** catches invalid parameters
+- **Required variables** are properly validated
+
+The test framework provides utilities from `~test` import to:
+
+- Initialize Terraform in the module directory
+- Apply Terraform with test variables
+- Validate the resulting infrastructure state
+- Test error conditions and edge cases
+
+Tests run in Docker with `--network=host` capabilities and help ensure module quality and reliability.
+
+### Documentation Requirements
+
+Every module requires a `README.md` with YAML frontmatter containing metadata like display name, description, icon, maintainer, and tags. The README must include:
+
+- **Purpose and functionality** - What the module does and when to use it
+- **Variable documentation** - All input variables with types, defaults, and descriptions  
+- **Resource listing** - What Coder resources the module creates
+- **Usage examples** - Basic and advanced configuration examples
+- **Requirements** - Any prerequisites or dependencies
+
+The documentation format is validated automatically and must pass validation before modules can be accepted. See the [Registry Contributing Guide](https://github.com/coder/registry/blob/main/CONTRIBUTING.md) for detailed formatting requirements and examples.
```

Comment:
I think this whole section can be removed or moved to Contributing docs in registry repo.